Source/control.cpp

```diff
@@ -264,17 +265,22 @@ void PrintInfo(const Surface &out)
 	const int LineStart[] = { 70, 58, 52, 48, 46 };
 	const int LineHeights[] = { 30, 24, 18, 15, 12 };
 
-	Rectangle line { GetMainPanel().position + Displacement { 177, LineStart[pnumlines] }, { 288, 12 } };
+	auto infoStringLines = (int)std::count(InfoString.str().begin(), InfoString.str().end(), '\n');
+
+	Rectangle line { GetMainPanel().position + Displacement { 177, LineStart[infoStringLines + pnumlines] }, { 288, 12 } };
 
 	if (!InfoString.empty()) {
-		DrawString(out, InfoString, line, InfoColor | UiFlags::AlignCenter | UiFlags::KerningFitSpacing, 2);
-		line.position.y += LineHeights[pnumlines];
+		for (string_view s : SplitByChar(InfoString, '\n')) {
+			DrawString(out, s, line, InfoColor | UiFlags::AlignCenter | UiFlags::KerningFitSpacing, 2);
+			line.position.y += LineHeights[pnumlines + infoStringLines];
+		}

```

Comment:
6th param of `DrawString()` is line height, is it not possible to use that instead of looping over `SplitByChar()`?

Reply:
Would I not still have to increment the y position for each line? How can I do that without looping?

Reply:
I replaced the loop and its contents with what you suggested, but the result is that I only see the first line of InfoString rendered, and the line height is incorrect.
![Screenshot_2022-09-07_18:46:12](https://user-images.githubusercontent.com/25830960/188997055-a41bc928-b603-4fc7-85d3-465a0e5529fb.png)
![Screenshot_2022-09-07_18:46:14](https://user-images.githubusercontent.com/25830960/188997061-7537c280-5588-4d35-9945-67dac13a0754.png)
![Screenshot_2022-09-07_18:46:16](https://user-images.githubusercontent.com/25830960/188997063-97a39f50-4318-4332-9049-7c88ae50d92c.png)

Reply:
Well the line height actually looks okay in 2/3

Reply:
Ok, I'll try to have a look at it later. Other then that I think things looks good now.

Reply:
I ran this through gdb and found that we're exiting early [here](https://github.com/diasurgical/devilutionX/blob/master/Source/engine/render/text_render.cpp#L386). The loop exits on the first newline.

Reply:
Check out the changes :smiley: